zerver/lib/slack_data_to_zulip_data.py

```diff
@@ -37,13 +37,79 @@ def get_model_id(model: Any, table_name: str, sequence_increase_factor: int) ->
+                             slack_data_dir: str) -> Tuple[ZerverFieldsT, AddedUsersT,
+                                                           AddedRecipientsT, AddedChannelsT]:
+    """
+    Returns:
+    1. realm, Converted Realm data
+    2. added_users, which is a dictionary to map from slack user id to zulip user id
+    3. added_recipient, which is a dictionary to map from channel name to zulip recipient_id
+    4. added_channels, which is a dictionary to map from channel name to zulip stream_id
+    """
+    # TODO fetch realm config from zulip config
+    DOMAIN_NAME = "zulipchat.com"
+    NOW = float(timezone_now().timestamp())
+
+    zerver_realm = get_zerver_realm(fixtures_path, REALM_ID, realm_subdomain, NOW)
+
+    realm = dict(zerver_client=[{"name": "populate_db", "id": 1},
+                                {"name": "website", "id": 2},
+                                {"name": "API", "id": 3}],
+                 zerver_userpresence=[],  # shows last logged in data, which is not available in slack
+                 zerver_userprofile_mirrordummy=[],
+                 zerver_realmdomain=[{"realm": REALM_ID,
+                                      "allow_subdomains": False,
+                                      "domain": DOMAIN_NAME,
+                                      "id": REALM_ID}],
+                 zerver_useractivity=[],
+                 zerver_realm=zerver_realm,
+                 zerver_huddle=[],
+                 zerver_userprofile_crossrealm=[],
+                 zerver_useractivityinterval=[],
+                 zerver_realmfilter=[],
+                 zerver_realmemoji=[])
+
+    zerver_userprofile, added_users = users_to_zerver_userprofile(slack_data_dir,
+                                                                  REALM_ID,
+                                                                  int(NOW),
+                                                                  DOMAIN_NAME)
+    channels_to_zerver_stream_fields = channels_to_zerver_stream(slack_data_dir,
+                                                                 REALM_ID,
+                                                                 added_users,
+                                                                 zerver_userprofile)
+    # See https://zulipchat.com/help/set-default-streams-for-new-users
+    # for documentation on zerver_defaultstream
+    realm['zerver_userprofile'] = zerver_userprofile
+
+    realm['zerver_defaultstream'] = channels_to_zerver_stream_fields[0]
+    realm['zerver_stream'] = channels_to_zerver_stream_fields[1]
+    realm['zerver_subscription'] = channels_to_zerver_stream_fields[3]
+    realm['zerver_recipient'] = channels_to_zerver_stream_fields[4]
+    added_channels = channels_to_zerver_stream_fields[2]
+    added_recipient = channels_to_zerver_stream_fields[5]
+
+    return realm, added_users, added_recipient, added_channels
+
+def get_zerver_realm(fixtures_path: str, REALM_ID: int, realm_subdomain: str,
```

Comment:
This should be named `create_zerver_realm`.

Reply:
But naming to such would conflict with the actual realm object creation function.

Reply:
`build_zerver_realm`?